index.js

```diff
@@ -361,13 +370,24 @@ function buildCode (location) {
     code += buildValue(propertyLocation, `obj[${JSON.stringify(key)}]`)
 
     const defaultValue = schema.properties[key].default
+    const constValue = schema.properties[key].const
+
     if (defaultValue !== undefined) {
       code += `
       } else {
         ${addComma}
-        json += ${asString} + ':' + ${JSON.stringify(JSON.stringify(defaultValue))}
+        json += ${asString} + ':' + ${JSON.stringify(
+        JSON.stringify(defaultValue)
+      )}
       `
-    } else if (required.includes(key)) {
+    } else if (isRequired && isConst) {
```

Comment:
This should work in a different way.
1. You should handle value serialization only in one place otherwise you will have code duplication and can make an error. In your case, you miss the null check here. So please handle all schema serialization logic inside buildValue function.
2. We don't need to do this check for const https://github.com/fastify/fast-json-stringify/blob/844eba5f4956230e1ec64bfaea18d0e595d066d4/index.js#L365.

Reply:
Null checks you mean this?

```javascript
const schema = {
  "type": "object",
  "properties": {
    "foo": {
      "const": "bar"
    }
  }
}


stringify({ "foo": "bar" }) // { "foo": "bar" }
stringify({ "foo": "baz" }) // { "foo": "bar" }
stringify({ "foo": 1 }) // { "foo": "bar" }
stringify({}) // { }
```

Reply:
To answer 2. 
If I don't have that checks how can I handle this?

```javascript
const schema = {
  "type": "object",
  "properties": {
    "foo": {
      "const": "bar"
    }
  }
}

stringify({}) // { }
```

Reply:
I think that the const value shouldn't work as a default value. It shouldn't have any other additional behavior.

```js
const schema = {
  "type": "object",
  "properties": {
    "foo": {
      "const": "bar"
    }
  },
  "required": ['foo']
}

stringify({}) // {}
```

Reply:
I am OK with that, but I remember that the majority agree on this [comment](https://github.com/fastify/fast-json-stringify/pull/505#issuecomment-1217390075).

Reply:
@Eomm @climba03003 I have some thoughts on why const shouldn't duplicate the default value. These are two different responsibilities. const is one of the keys that determine what values a property can take aka type/set of value. It shouldn't have any other behavioral side effects.

I will give you an example. I can define an integer property with constraints min: 1 and max: 1. The only possible value is 1, but it doesn't mean that I should serialize this property if user doesn't pass the value.

Reply:
> I think that the const value shouldn't work as a default value. It shouldn't have any other additional behavior.
> 
> ```js
> const schema = {
>   "type": "object",
>   "properties": {
>     "foo": {
>       "const": "bar"
>     }
>   },
>   "required": ['foo']
> }
> 
> stringify({}) // {}
> ```

@ivan-tymoshenko Why that should not raise the error `"foo" is required!` ?

Reply:
Yes, you are right, it should. From my point of view.

Reply:
If you are waiting for me, I'm leaving this to you.

Reply:
I would like to hear @climba03003 opinion as it was his suggestion https://github.com/fastify/fast-json-stringify/pull/505#issuecomment-1217390075.

Reply:
When the schema said the property is required, and missing one from the input.
It just convenient to always put the value there since it is `required`.

Just like why we always put the `const` value regardless the input.

Reply:
I do not block if it turns to throw.
It consistence with the current `required` handling for other properties.